analyzers/src/SonarAnalyzer.CSharp/Rules/InterfacesShouldNotBeEmpty.cs

```diff
@@ -42,14 +42,29 @@ public sealed class InterfacesShouldNotBeEmpty : SonarDiagnosticAnalyzer
 
                     var interfaceSymbol = c.SemanticModel.GetDeclaredSymbol(interfaceDeclaration);
                     if (interfaceSymbol is { DeclaredAccessibility: Accessibility.Public }
-                        && !IsAggregatingOtherInterfaces(interfaceSymbol))
+                        && !IsAggregatingOtherInterfaces(interfaceSymbol)
+                        && !IsSpecializedGeneric(interfaceSymbol)
+                        && !HasEnhancingAttribute(interfaceSymbol))
                     {
                         c.ReportIssue(Diagnostic.Create(Rule, interfaceDeclaration.Identifier.GetLocation()));
                     }
                 },
                 SyntaxKind.InterfaceDeclaration);
-
         private static bool IsAggregatingOtherInterfaces(ITypeSymbol interfaceSymbol) =>
             interfaceSymbol.AllInterfaces.Length > 1;
+
+        private static bool IsSpecializedGeneric(INamedTypeSymbol interfaceSymbol) =>
+            !interfaceSymbol.Interfaces.IsEmpty && (IsBoundGeneric(interfaceSymbol) || IsConstraintGeneric(interfaceSymbol));
```

Comment:
If the one above turns "AllInterfaces" into "Interfaces" I guess we can have a `ImplementsOneInterface` method instead of `IsAggregatingOtherInterfaces` and simplify this one to include the 2 OR statements only

Reply:
The difference is still ` interfaceSymbol.Interfaces.Length > 1` means at least 2 interfaces, while `!interfaceSymbol.Interfaces.IsEmpty` means at least 1 interface.